packages/devextreme/js/ui/drop_down_button.js

```diff
@@ -117,14 +115,23 @@ const DropDownButton = Widget.inherit({
         this._createItemClickAction();
         this._createActionClickAction();
         this._createSelectionChangedAction();
-        this._initDataSource();
+        this._initDataController();
         this._compileKeyGetter();
         this._compileDisplayGetter();
-        this._itemsToDataSource(this.option('items'));
         this._options.cache('buttonGroupOptions', this.option('buttonGroupOptions'));
         this._options.cache('dropDownOptions', this.option('dropDownOptions'));
     },
 
+    _initDataController() {
+        const dataSource = this.option('dataSource');
+
+        if(dataSource) {
+            this._dataController = new DataController(dataSource);
+        } else {
+            this._dataController = DataController.initFromArray(this.option('items'), this._getKey());
+        }
```

Comment:
Don't we want to incapsulate this logic inside of `dataController`? It's actual not only for a DropDownButton.

Reply:
I can't see a smooth solution to move this logic into the DataController. Make 3 optional parameters in constructor? Its possible but in this case we will always perform unnecessary getters - this.option('items') and this._getKey().

Reply:
I believe we're supposed to do it with a static method at all.
`dataSource` option can be a usual array too => we should handle this on dataSource creating inside a DataController.
I think it's okay to pass `dataSource || items` as a first parameter + add one more optional parameter `config`